### PR TITLE
CRUD for tasks

### PR DESCRIPTION
**Pull request by adarsh**

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  def new
+    @task = Task.new
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def create
+    @task = Task.new(params[:task])
```
Comment (harlow) [inline app/controllers/tasks_controller.rb]: use strong params. `@task = Task.new(task_params)`

File: app/views/tasks/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<h1>Todo Manager</h1>
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th></th>
+    <th></th>
+  </tr>
+
+<% @tasks.each do |task| %>
```
Comment (harlow) [inline app/views/tasks/index.html.erb]: create a partial for `_task.html.erb` and then `render @tasks`

File: spec/features/visit_the_homepage_spec.rb [hunk 3]
```diff
@@ -1,9 +1,41 @@
 require 'spec_helper'
```
Comment (harlow) [inline spec/features/visit_the_homepage_spec.rb]: rename this file `user_manages_todos_spec.rb`

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class TasksController < ApplicationController
+  respond_to :html
```
Comment (harlow) [inline app/controllers/tasks_controller.rb]: move this to `application_controller.rb`

File: app/models/task.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class Task < ActiveRecord::Base
+  include ActiveModel::ForbiddenAttributesProtection
```
Comment (harlow) [inline app/models/task.rb]: move this to and initializer in `config/initializers/strong_parameters.rb`

``` ruby
ActiveRecord::Base.send :include, ActiveModel::ForbiddenAttributesProtection
```

File: app/views/tasks/index.html.erb (hunk 2 above)
Comment (adarsh) [inline app/views/tasks/index.html.erb]: Extracted 03fb7fe

File: spec/features/visit_the_homepage_spec.rb [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+
+    visit root_path
+    click_link 'New Task'
+    fill_in 'Name', with: todo_description
+    click_button('Create Task')
+
+    expect(page).to have_content(todo_description)
+  end
+
+  scenario 'user sees tasks in the index' do
```
Comment (harlow) [inline spec/features/visit_the_homepage_spec.rb]: this test is duplicated.

File: spec/features/visit_the_homepage_spec.rb (hunk 3 above)
Comment (adarsh) [inline spec/features/visit_the_homepage_spec.rb]: Renamed 6a71a33

File: app/controllers/tasks_controller.rb (hunk 4 above)
Comment (adarsh) [inline app/controllers/tasks_controller.rb]: moved 1f7a8cb

File: app/models/task.rb (hunk 5 above)
Comment (adarsh) [inline app/models/task.rb]: moved 5d0592d

File: app/views/tasks/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th></th>
+    <th></th>
+  </tr>
+
+  <%= render @tasks %>
+
```
Comment (harlow) [inline app/views/tasks/index.html.erb]: remove line break

File: spec/features/user_manages_todos_spec.rb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    fill_in 'Name', with: todo_description
+    click_button('Create Task')
+    click_link 'Back'
+
+    expect(page).to have_content(todo_description)
+  end
+
+  scenario 'user edits the task' do
+    original_name = 'Mow the lawn'
+    new_name = 'Paint the fence'
```
Comment (harlow) [inline spec/features/user_manages_todos_spec.rb]: change `new_name` to `updated_name`

File: app/views/tasks/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        <li><%= msg %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field">
+    <%= f.label :name %><br />
+    <%= f.text_field :name %>
+  </div>
```
Comment (harlow) [inline app/views/tasks/_form.html.erb]: line break

File: app/views/tasks/_form.html.erb [hunk 10]
```diff
@@ -0,0 +1,21 @@
+<%= form_for(@task) do |f| %>
+  <% if @task.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@task.errors.count, "error") %> prohibited this task from being saved:</h2>
+
+      <ul>
+      <% @task.errors.full_messages.each do |msg| %>
```
Comment (harlow) [inline app/views/tasks/_form.html.erb]: indentation.

File: app/views/tasks/_form.html.erb [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<%= form_for(@task) do |f| %>
+  <% if @task.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@task.errors.count, "error") %> prohibited this task from being saved:</h2>
```
Comment (harlow) [inline app/views/tasks/_form.html.erb]: double quotes

File: spec/models/task_spec.rb [hunk 12]
```diff
@@ -0,0 +1,4 @@
+require 'spec_helper'
+
+describe Task, 'validations' do
```
Comment (harlow) [inline spec/models/task_spec.rb]: remove empty test

**Pull request merged**